### ISSUE TITLE
Make it possible to handle orders that are rejected by the validate callback

### ISSUE THREAD

**Issue by erik-kallen** [klarna/kco-mobile-sdk#67]
**Is your feature request related to a problem? Please describe.**

The KlarnaCheckout API supports specifying a validate callback (https://docs.klarna.com/klarna-checkout/popular-use-cases/validate-order/). This callback is supposed to return either `200 OK` (to continue with the order), or `303 See Other` (if the order is invalid).

When the callback returns `303 See Other`, this SDK will unconditionally open a browser with that URL. However, in my app I want to show the validation errors inline in the app instead.

The `merchantHandlesValidationErrors` flag seems to have no effect on this behavior.

**Describe the solution you'd like**

I want a new flag `merchantHandlesOrderValidationErrors` that makes the SDK raises a notification `order_validation_error` instead of popping up a new window. The Location header returned by the validate call should be sent as data to the notification.

**Describe alternatives you've considered**
None

**Comment by erik-kallen** [comment]
It seems that this does not need fixing after all.

The problem was that I set `self.checkout.merchantHandlesValidationErrors = YES` immediately after creating `self.checkout`. When I moved the assignment to immediately before I do `[self.checkout setSnippet: ..]` it magically started working.

I'm leaving it open because the documentation only says that the property needs to be set before starting the Checkout process, but there is obviously some other requirement as well. Additionally, the documentation does not at all say how the validation error is signaled (appears to be `{ error_type: "merchant_handles_validation_error", url: "https://whatever" }`
